src/create_disk.sh

```diff
@@ -472,32 +487,50 @@ generate_gpgkeys() {
     rm -rf "${tmp_home}"
 }
 
+install_bootloader_nons390x_bootupd_epoch_0() {
+    case $arch in
+    x86_64)
+        install_uefi_bootupd_epoch_0
+        if [ "${x86_bios_bootloader}" = 1 ]; then
+            # And BIOS grub in addition.  See also
+            # https://github.com/coreos/fedora-coreos-tracker/issues/32
+            # Install BIOS/PReP bootloader using the target system's grub2-install,
+            # see https://github.com/coreos/coreos-assembler/issues/3156
+            chroot_run /sbin/grub2-install \
+                --target i386-pc \
+                --boot-directory $rootfs/boot \
+                --modules mdraid1x \
+                "$disk"
+        fi
+        ;;
+    aarch64)
+        # Our aarch64 is UEFI only.
+        install_uefi_bootupd_epoch_0
+        ;;
+    ppc64le)
+        # to populate PReP Boot, i.e. support pseries
+        chroot_run /sbin/grub2-install --target=powerpc-ieee1275 --boot-directory $rootfs/boot --no-nvram "${disk}${PREPPN}"
+        install_grub_cfg
+        ;;
+    *) "unhandled arch $arch 1>&2"; exit 1
+        ;;
+    esac
+}
+
 # Other arch-specific bootloader changes
 # shellcheck disable=SC2031
 case "$arch" in
-x86_64)
-    # UEFI
-    install_uefi
-    if [ "${x86_bios_bootloader}" = 1 ]; then
-        # And BIOS grub in addition.  See also
-        # https://github.com/coreos/fedora-coreos-tracker/issues/32
-        # Install BIOS/PReP bootloader using the target system's grub2-install,
-        # see https://github.com/coreos/coreos-assembler/issues/3156
-        chroot_run /sbin/grub2-install \
-            --target i386-pc \
-            --boot-directory $rootfs/boot \
-            --modules mdraid1x \
-            "$disk"
-    fi
-    ;;
-aarch64)
-    # Our aarch64 is UEFI only.
-    install_uefi
-    ;;
-ppc64le)
-    # to populate PReP Boot, i.e. support pseries
-    chroot_run /sbin/grub2-install --target=powerpc-ieee1275 --boot-directory $rootfs/boot --no-nvram "${disk}${PREPPN}"
-    install_grub_cfg
+x86_64|aarch64|ppc64le)
+    case "${bootupd_epoch}" in
+        0)
+            install_bootloader_nons390x_bootupd_epoch_0
+            ;;
+        1)
+            install_via_bootupd
```

Comment:
Overall looks good to me, one minor comment, I think you meant `bootloader_bootupd_epoch_1`?

Reply:
Looks like it to me too.